tests/test_objects.py

```diff
@@ -573,11 +573,11 @@ def test_merge_no_link_list():
     [
         (
             (
-                {"fruits": ["apple"], "vegetables": ["beet"]},
-                {"fruits": ["banana"], "vegetables": ["carrot"]},
+                {"fruits": ["apple"], "ohters": { "vegetables": ["beet"]}},
+                {"fruits": ["banana"], "ohters": { "vegetables": ["carrot"]}},
                 lambda a, b: a + b if isinstance(a, list) else b,
```

Comment:
This test fails with this iteratee but can be made to pass by changing `else b` to `else None`. When `None` is returned by a custom iteratee, the `None` is ignored and the merge will recurse if conditions are met. But when a non-`None` value is returned, the recursion won't happen. I was able to get the tests passing locally with `lambda a, b: a + b if isinstance(a, list) else None`.